sunpy/util/util.py

```diff
@@ -212,3 +212,28 @@ def expand_list_generator(inp):
                 yield nested_item
         else:
             yield item
+
+def partial_match(key, dictionary):
+    """
+    Return value/values based on the partial key .
```

Comment:
This needs a bit more explanation. Also the space between `key` and `.` should be removed.